<launch>
  <arg name="folder_path"            default="$(find grid_map_pcl)/config"/>
  <arg name="pcd_filename"           default="UMA.pcd" />
  <arg name="config_file_path"       default="$(find grid_map_pcl)/config/parameters.yaml" />
  <arg name="map_rosbag_topic"       default="grid_map" />
  <arg name="output_grid_map"        default="$(find grid_map_pcl)/data/input_cloud.bag" />
  <arg name="map_frame"              default="map" /> <!-- map -->
  <arg name="map_layer_name"         default="elevation" />
  <arg name="prefix"                 default=""/>
  <arg name="set_verbosity_to_debug" default="false"/>
  <arg name="publish_point_cloud"    default="true"/>
    
  <node name="grid_map_pcl_loader_node" 
        pkg="grid_map_pcl"
        type="grid_map_pcl_loader_node" 
        output="screen" 
        launch-prefix="$(arg prefix)">
    <rosparam file="$(arg config_file_path)" />
    <param name="config_file_path"        type="string" value="$(arg config_file_path)" />
    <param name="folder_path"             type="string" value="$(arg folder_path)" />
    <param name="pcd_filename"            type="string" value="$(arg pcd_filename)" />
    <param name="map_rosbag_topic"        type="string" value="$(arg map_rosbag_topic)" />
    <param name="output_grid_map"         type="string" value="$(arg output_grid_map)" />
    <param name="map_frame"               type="string" value="$(arg map_frame)" />
    <param name="map_layer_name"          type="string" value="$(arg map_layer_name)" />
    <param name="set_verbosity_to_debug"  type="bool"   value="$(arg set_verbosity_to_debug)" />
  </node>
  
  <!-- <node name="rviz_grid_map_pcl" pkg="rviz" type="rviz" args="-d $(find grid_map_pcl)/rviz/grid_map_vis.rviz" /> -->
  
  <node name="point_cloud_publisher_node" 
        pkg="grid_map_pcl"
        type="pointcloud_publisher_node"
        output="screen"
        launch-prefix="$(arg prefix)"
        if="$(arg publish_point_cloud)">
    <param name="folder_path"             type="string" value="$(arg folder_path)" />
    <param name="pcd_filename"            type="string" value="$(arg pcd_filename)" />
    <param name="cloud_frame"             type="string" value="$(arg map_frame)" />
  </node>

    <!-- Launch the grid map filters node. -->
  <node pkg="filtros_carlos" type="filtro_carlos_pre" name="Filtro" output="screen">
    <!-- Input topic -->
  	<param name="input_topic" value="/grid_map_pcl_loader_node/grid_map_from_raw_pointcloud" />
  	<!-- Output topic -->
  	<param name="output_topic" value="filtro_mapa_pre" />
  	<!-- Load grid map filter chain configuration -->
    <rosparam command="load" file="$(find filtros_carlos)/config/filtro_preprocesado.yaml" />
  </node>

    <!-- Launch the grid map visualizer -->
  <node pkg="grid_map_visualization" type="grid_map_visualization" name="grid_map_visualization" output="screen">
    <rosparam command="load" file="$(find filtros_carlos)/config/Visualizacion.yaml" />
  </node>

    <!-- Launch RViz with the demo configuration -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find filtros_carlos)/rviz/filtro.rviz" />

</launch>